packages/wdio-shared-store-service/tests/client.test.ts

```diff
@@ -41,6 +41,12 @@ describe('client', () => {
     })
 
     describe('when used in launcher process', () => {
+        it('Error should be thrown when attempting to retrieve a value before server initialization', () => {
+            expect(getValue('*')).rejects.toThrowError('Attempting to use `getValue` before the server has been initialized.')
+            expect(getValueFromPool('*')).rejects.toThrowError('Attempting to use `getValueFromPool` before the server has been initialized.')
+            expect(addValueToPool('*', '')).rejects.toThrowError('Attempting to use `addValueToPool` before the server has been initialized.')
```

Comment:
This is not retrieving, put it in a separate testcase?

Reply:
Considering that these are negative tests and should run before invoking `setPort(port)`, which is responsible for resolving the value of `isBaseUrlReady`, I've made this changes. Please share your feedback on this. If necessary, I'm open to move this to a new test case.

Reply:
I think it makes sense this way.